internal/handlers/pg/pgdb/settings.go

```diff
@@ -0,0 +1,226 @@
+func (pgPool *Pool) CreateSettingsTable(ctx context.Context, db string) error {
+	tables, err := pgPool.tables(ctx, db)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	if slices.Contains(tables, settingsTableName) {
+		return ErrAlreadyExist
+	}
+
+	sql := `CREATE TABLE ` + pgx.Identifier{db, settingsTableName}.Sanitize() + ` (settings jsonb)`
+	_, err = pgPool.Exec(ctx, sql)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	settings := must.NotFail(types.NewDocument("collections", must.NotFail(types.NewDocument())))
+	sql = fmt.Sprintf(`INSERT INTO %s (settings) VALUES ($1)`, pgx.Identifier{db, settingsTableName}.Sanitize())
+	_, err = pgPool.Exec(ctx, sql, must.NotFail(fjson.Marshal(settings)))
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	return nil
+}
+
+// GetTableName returns the name of the table for given collection or error.
+func (pgPool *Pool) GetTableName(ctx context.Context, db, collection string) (string, error) {
+	var err error
+
+	schemaExists, err := pgPool.schemaExists(ctx, db)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	if !schemaExists {
+		return GetTableNameFormatted(collection), nil
+	}
+
+	var tables []string
+	if tables, err = pgPool.tables(ctx, db); err != nil {
+		return "", lazyerrors.Error(err)
+	}
+	if !slices.Contains(tables, settingsTableName) {
+		err = pgPool.CreateSettingsTable(ctx, db)
+		if err != nil {
+			return "", lazyerrors.Error(err)
+		}
+	}
+
+	tx, err := pgPool.Begin(ctx)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+	defer func() {
+		if err != nil {
+			must.NoError(tx.Rollback(ctx))
+			return
+		}
+		must.NoError(tx.Commit(ctx))
+	}()
+
+	settings, err := pgPool.getSettingsTable(ctx, tx, db)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	collectionsDoc := must.NotFail(settings.Get("collections"))
+	collections, ok := collectionsDoc.(*types.Document)
+	if !ok {
+		return "", lazyerrors.Errorf("expected document but got %[1]T: %[1]v", collectionsDoc)
+	}
+
+	if collections.Has(collection) {
+		return must.NotFail(collections.Get(collection)).(string), nil
+	}
+
+	tableName := GetTableNameFormatted(collection)
+	must.NoError(collections.Set(collection, tableName))
+	must.NoError(settings.Set("collections", collections))
+
+	err = pgPool.updateSettingsTable(ctx, tx, db, settings)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	return tableName, nil
+}
+
+// updateSettingsTable updates FerretDB settings table.
+func (pgPool *Pool) updateSettingsTable(ctx context.Context, tx pgx.Tx, db string, settings *types.Document) error {
+	sql := `UPDATE ` + pgx.Identifier{db, settingsTableName}.Sanitize() + `SET settings = $1`
+	_, err := tx.Exec(ctx, sql, must.NotFail(fjson.Marshal(settings)))
+	return err
+}
+
+// getSettingsTable returns FerretDB settings table.
+func (pgPool *Pool) getSettingsTable(ctx context.Context, tx pgx.Tx, db string) (*types.Document, error) {
```

Comment:
The type of returning value is `types.Document`. Why / what for? 
*  It's not convenient as you later decomposing the document and receiving from `must.Get`
* * also it's a `pgdb` handler's package, it can use bare values, isn't it?